### PR TITLE
feat: remix deployment

### PR DESCRIPTION
**Pull request by WillHutt**

### CODE REVIEW

**Comment by tvanantwerp** [comment]
Noticing a _lot_ of files with changes like this, despite `.prettierrc` clearly saying `singleQuote` is preferred. Seems like most changed files in this PR are just single => double quotes, going against the `.prettierrc`.

I'm going to look and see if any of the other examples have husky set up with lint-staged—I think that would be useful.

image

File: remix-gql-tailwind/app/components/FileExplorer/FileExplorer.view.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import type { ExplorerItem } from './types';
```
Comment (danecando) [inline remix-gql-tailwind/app/components/FileExplorer/FileExplorer.view.tsx:1]: It looks like prettier is set to single quote but these changed to double. Double check on that you might need to run prettier

File: remix-gql-tailwind/remix.config.js [hunk 2]
```diff
@@ -1,11 +1,25 @@
...
  */
 module.exports = {
   appDirectory: "app",
+  cacheDirectory: "./node_modules/.cache/remix",
   assetsBuildDirectory: "public/build",
-  publicPath: "/build/",
-  serverBuildDirectory: "build",
-  devServerPort: 8002,
-  ignoredRouteFiles: [".*"],
+  publicPath: "/_static/build/",
```
Comment (danecando) [inline remix-gql-tailwind/remix.config.js:10]: publicPath need to be changed like this for the arc deploy setup?

File: remix-gql-tailwind/vitest.config.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import { defineConfig } from "vite";
+import react from "@vitejs/plugin-react";
+import tsconfigPaths from "vite-tsconfig-paths";
+
+export default defineConfig({
+  plugins: [react(), tsconfigPaths()],
+  test: {
+    globals: true,
+    environment: "happy-dom",
```
Comment (danecando) [inline remix-gql-tailwind/vitest.config.ts:12]: just a note I tried using happy-dom once and had some issues with some of the event apis from testing library if you ever notice an issue there

File: remix-gql-tailwind/server.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { createRequestHandler } from "@remix-run/architect";
+import * as build from "@remix-run/dev/server-build";
+
+// if (process.env.NODE_ENV !== "production") {
+//   require("./mocks");
+// }
```
Comment (danecando) [inline remix-gql-tailwind/server.ts:6]: is there a reason this is just commented out for now vs just removing it if it's not needed?

**Review by danecando — CHANGES_REQUESTED** [review]
Main thing is just making sure the formatting is correct with prettier. Doesn't look like a lot actually changed besides some WIP cypress stuff and the deployment stuff

File: remix-gql-tailwind/.github/workflows/remix-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
+name: Remix GQL Tailwind
```
[the remark this answers is not in the document]
Comment (WillHutt) [reply remix-gql-tailwind/.github/workflows/remix-deploy.yml:1]: I'm not sure what you mean. Though we have a number of different workflows in our main and they also have names.

File: remix-gql-tailwind/remix.config.js (hunk 2 above)
Comment (WillHutt) [reply remix-gql-tailwind/remix.config.js:10]: Yes, at least based on the grunge stack and docs I've seen.

File: remix-gql-tailwind/vitest.config.ts (hunk 3 above)
Comment (WillHutt) [reply remix-gql-tailwind/vitest.config.ts:12]: Good to know. Making a note of this.

File: remix-gql-tailwind/server.ts (hunk 4 above)
Comment (WillHutt) [reply remix-gql-tailwind/server.ts:6]: It's currently commented out as we don't have `./mocks` fleshed out and I don't want to miss it. Let me add a TODO for clarity.

File: remix-gql-tailwind/.github/workflows/remix-deploy.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (danecando) [reply remix-gql-tailwind/.github/workflows/remix-deploy.yml:1]: I just assumed github workflows had to be in the root of the repo

[the remark this answers is not in the document]
Comment (WillHutt) [reply remix-gql-tailwind/.github/workflows/remix-deploy.yml:1]: Ahh 🤦 I see what you mean. Let me fix that.

**Review by danecando — APPROVED** [review]

**Pull request merged**